addons/sale_mrp/models/sale.py

```diff
@@ -15,11 +15,21 @@ class SaleOrder(models.Model):
     @api.depends('procurement_group_id.stock_move_ids.created_production_id.procurement_group_id.mrp_production_ids')
     def _compute_mrp_production_count(self):
         for sale in self:
-            sale.mrp_production_count = len(sale.procurement_group_id.stock_move_ids.created_production_id.procurement_group_id.mrp_production_ids)
+            production_moves = 0
+            moves = sale.procurement_group_id.stock_move_ids
+            for move in moves:
+                while move.move_orig_ids and not move.created_production_id:
+                    move = move.move_orig_ids
+                production_moves += len(move.created_production_id.procurement_group_id.mrp_production_ids)
+            sale.mrp_production_count = production_moves
```

Comment:
I would create a set of production orders then call `len()` at the very end to avoid duplication

Reply:
Ok, indeed it is safer this way, thank you for the review, I updated the code